pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MultistageGroupByExecutor.java

```diff
@@ -241,6 +241,11 @@ private void processAggregate(TransferableBlock block) {
           aggFunction.aggregateGroupBySV(numMatchedRows, filteredIntKeys, groupByResultHolder, blockValSetMap);
         }
       }
+      if (intKeys == null) {
+        // _groupIdGenerator should still have all the groups even if there are only filtered aggregates for SQL
+        // compliant results.
+        generateGroupByKeys(block);
+      }
```

Comment:
Wouldn't be semantically the same to always initialize `intKeys` to `generateGroupByKeys(block);` in line 209?

Reply:
Yup, this is actually left over from my earlier choice of making it optional for the multi-stage engine too. I can move it back if we decide we're okay with keeping this the default behavior for the multi-stage engine.